app.js

```diff
@@ -1,12 +1,31 @@
 const express = require('express')
-const { getEndpoints, getTopics } = require('./controllers/app.controller')
+const { getEndpoints, getTopics, getArticles } = require('./controllers/app.controller')
+const { commentData } = require('./db/data/test-data')
 
 const app = express()
 
 app.get('/api', getEndpoints)
 
 app.get('/api/topics', getTopics)
 
+app.get('/api/articles/:article_id', getArticles)
+
+app.use((err, request, response, next) => {
+    if (err.code === '22P02') {
+        response.status(400).send({msg: 'bad request'})
+    } else {
+        next(err)
+    }
+})
+
+app.use((err, request, response, next) => {
+    if (err.status === 404) {
+        response.status(404).send({msg: 'not found'})
+    } else {
+        next(err)
```

Comment:
This is extremely close. Best advice I can give you is to look at the bottom of the error handling notes where there is an example of error functions extracted out into individual handlers. It's very close, I'd move it above the psql handler and consider changing the if statement to something like ```if(err.status && err.msg)``` That way this custom handler is more reusable.